Library/Homebrew/cleaner.rb

```diff
@@ -29,9 +29,31 @@ def clean
     [@f.bin, @f.sbin, @f.lib].each { |d| clean_dir(d) if d.exist? }
 
     # Get rid of any info 'dir' files, so they don't conflict at the link stage
+    #
+    # The 'dir' files come in at least 3 locations:
+    #
+    # 1. 'info/dir'
+    # 2. 'info/#{name}/dir'
+    # 3. 'info/#{arch}/dir'
+    #
+    # Of these 3 only 'info/#{name}/dir' is safe to keep since the rest will
+    # conflict with other formulae because they use a shared location.
+    #
+    # See [cleaner: recursively delete info `dir`s by gromgit · Pull Request
+    # #11597][1], [emacs 28.1 bottle does not contain `dir` file · Issue
+    # #100190][2], and [Keep `info/#{f.name}/dir` files in cleaner by
+    # timvisher][3] for more info.
+    #
+    # [1]: https://github.com/Homebrew/brew/pull/11597
+    # [2]: https://github.com/Homebrew/homebrew-core/issues/100190
+    # [3]: https://github.com/Homebrew/brew/pull/13215
```

Comment:
@MikeMcQuaid What do you think? It doesn't seems like we try to stay overly concise in most of the comments in this file but maybe this is a bit too flowery for what you were going for?

Reply:
No, this is absolutely wonderful, nice work.